### ISSUE TITLE
Inifinite loop in Debug fmt

### ISSUE THREAD

**Issue by kornelski** [rust-lang/rust-analyzer#16936]
After a few queries rust-analyzer gets stuck using 100% CPU all the time, and all of that time is spent appending to a string in an `fmt::Debug` implementation.

rust-analyzer 1.79.0-nightly (2f090c30 2024-03-23)

```text
294.03 Gc   40.3%	294.03 Gc	 	 _$LT$str$u20$as$u20$core..fmt..Debug$GT$::fmt::hd9e26fc12c3bb048
294.03 Gc   40.3%	-	 	  core::fmt::write::he4d5fa2daff1f531
294.03 Gc   40.3%	-	 	   _$LT$hir_ty..mir..eval..MirEvalError$u20$as$u20$core..fmt..Debug$GT$::fmt::h4d93694127b4a6b8
294.03 Gc   40.3%	-	 	    core::fmt::builders::DebugStruct::field::h84442bdb6c623fac
294.03 Gc   40.3%	-	 	     _$LT$hir_ty..mir..eval..MirEvalError$u20$as$u20$core..fmt..Debug$GT$::fmt::h4d93694127b4a6b8
294.03 Gc   40.3%	-	 	      core::fmt::builders::DebugStruct::field::h84442bdb6c623fac
294.03 Gc   40.3%	-	 	       _$LT$hir_ty..mir..eval..MirEvalError$u20$as$u20$core..fmt..Debug$GT$::fmt::h4d93694127b4a6b8
294.03 Gc   40.3%	-	 	        core::fmt::builders::DebugStruct::field::h84442bdb6c623fac
294.03 Gc   40.3%	-	 	         _$LT$hir_ty..mir..eval..MirEvalError$u20$as$u20$core..fmt..Debug$GT$::fmt::h4d93694127b4a6b8
294.03 Gc   40.3%	-	 	          core::fmt::write::he4d5fa2daff1f531
294.03 Gc   40.3%	-	 	           alloc::fmt::format::format_inner::h8378d1c89f1ce828
163.41 Gc   22.4%	163.41 Gc	 	 alloc::string::String::push::h051761fb598e9e2d
123.63 Gc   16.9%	-	 	  _$LT$alloc..string..String$u20$as$u20$core..fmt..Write$GT$::write_char::h289735d11ee72003
123.63 Gc   16.9%	-	 	   _$LT$str$u20$as$u20$core..fmt..Debug$GT$::fmt::hd9e26fc12c3bb048
123.63 Gc   16.9%	-	 	    core::fmt::write::he4d5fa2daff1f531
123.63 Gc   16.9%	-	 	     _$LT$hir_ty..mir..eval..MirEvalError$u20$as$u20$core..fmt..Debug$GT$::fmt::h4d93694127b4a6b8
123.63 Gc   16.9%	-	 	      core::fmt::builders::DebugStruct::field::h84442bdb6c623fac
123.63 Gc   16.9%	-	 	       _$LT$hir_ty..mir..eval..MirEvalError$u20$as$u20$core..fmt..Debug$GT$::fmt::h4d93694127b4a6b8
123.63 Gc   16.9%	-	 	        core::fmt::builders::DebugStruct::field::h84442bdb6c623fac
123.63 Gc   16.9%	-	 	         _$LT$hir_ty..mir..eval..MirEvalError$u20$as$u20$core..fmt..Debug$GT$::fmt::h4d93694127b4a6b8
123.63 Gc   16.9%	-	 	          core::fmt::builders::DebugStruct::field::h84442bdb6c623fac
123.63 Gc   16.9%	-	 	           _$LT$hir_ty..mir..eval..MirEvalError$u20$as$u20$core..fmt..Debug$GT$::fmt::h4d93694127b4a6b8
123.63 Gc   16.9%	-	 	            core::fmt::write::he4d5fa2daff1f531
123.63 Gc   16.9%	-	 	             alloc::fmt::format::format_inner::h8378d1c89f1ce828
39.78 Gc    5.4%	-	 	  _$LT$str$u20$as$u20$core..fmt..Debug$GT$::fmt::hd9e26fc12c3bb048
39.78 Gc    5.4%	-	 	   core::fmt::write::he4d5fa2daff1f531
39.78 Gc    5.4%	-	 	    _$LT$hir_ty..mir..eval..MirEvalError$u20$as$u20$core..fmt..Debug$GT$::fmt::h4d93694127b4a6b8
39.78 Gc    5.4%	-	 	     core::fmt::builders::DebugStruct::field::h84442bdb6c623fac
39.78 Gc    5.4%	-	 	      _$LT$hir_ty..mir..eval..MirEvalError$u20$as$u20$core..fmt..Debug$GT$::fmt::h4d93694127b4a6b8
39.78 Gc    5.4%	-	 	       core::fmt::builders::DebugStruct::field::h84442bdb6c623fac
39.78 Gc    5.4%	-	 	        _$LT$hir_ty..mir..eval..MirEvalError$u20$as$u20$core..fmt..Debug$GT$::fmt::h4d93694127b4a6b8
39.78 Gc    5.4%	-	 	         core::fmt::builders::DebugStruct::field::h84442bdb6c623fac
39.78 Gc    5.4%	-	 	          _$LT$hir_ty..mir..eval..MirEvalError$u20$as$u20$core..fmt..Debug$GT$::fmt::h4d93694127b4a6b8
39.78 Gc    5.4%	-	 	           core::fmt::write::he4d5fa2daff1f531
39.78 Gc    5.4%	-	 	            alloc::fmt::format::format_inner::h8378d1c89f1ce828
118.61 Gc   16.2%	118.61 Gc	 	 _$LT$alloc..string..String$u20$as$u20$core..fmt..Write$GT$::write_str::h6da2d64d3abbdf5c
```

There's also some suspiciously deep recursion, although not infinite, because it doesn't crash:

```text
39.91 Gc    5.4%	-	 	  core::fmt::write::he4d5fa2daff1f531
39.91 Gc    5.4%	-	 	   _$LT$hir_ty..mir..eval..MirEvalError$u20$as$u20$core..fmt..Debug$GT$::fmt::h4d93694127b4a6b8
39.91 Gc    5.4%	-	 	    core::fmt::builders::DebugStruct::field::h84442bdb6c623fac
39.91 Gc    5.4%	-	 	     _$LT$hir_ty..mir..eval..MirEvalError$u20$as$u20$core..fmt..Debug$GT$::fmt::h4d93694127b4a6b8
39.91 Gc    5.4%	-	 	      core::fmt::builders::DebugStruct::field::h84442bdb6c623fac
39.91 Gc    5.4%	-	 	       _$LT$hir_ty..mir..eval..MirEvalError$u20$as$u20$core..fmt..Debug$GT$::fmt::h4d93694127b4a6b8
39.91 Gc    5.4%	-	 	        core::fmt::builders::DebugStruct::field::h84442bdb6c623fac
39.91 Gc    5.4%	-	 	         _$LT$hir_ty..mir..eval..MirEvalError$u20$as$u20$core..fmt..Debug$GT$::fmt::h4d93694127b4a6b8
39.91 Gc    5.4%	-	 	          core::fmt::write::he4d5fa2daff1f531
39.91 Gc    5.4%	-	 	           alloc::fmt::format::format_inner::h8378d1c89f1ce828
39.91 Gc    5.4%	-	 	            hir_ty::mir::eval::shim::_$LT$impl$u20$hir_ty..mir..eval..Evaluator$GT$::detect_and_exec_special_function::h6adaafd39ac2745b
39.91 Gc    5.4%	-	 	             hir_ty::mir::eval::Evaluator::exec_fn_with_args::h5acbc73e38ea6c7e
39.91 Gc    5.4%	-	 	              hir_ty::mir::eval::Evaluator::exec_looked_up_function::hdd39238ca1e7535e
39.91 Gc    5.4%	-	 	               hir_ty::mir::eval::Evaluator::exec_fn_with_args::h5acbc73e38ea6c7e
39.91 Gc    5.4%	-	 	                hir_ty::mir::eval::shim::_$LT$impl$u20$hir_ty..mir..eval..Evaluator$GT$::detect_and_exec_special_function::h6adaafd39ac2745b
39.91 Gc    5.4%	-	 	                 hir_ty::mir::eval::Evaluator::exec_fn_with_args::h5acbc73e38ea6c7e
39.91 Gc    5.4%	-	 	                  hir_ty::mir::eval::Evaluator::exec_looked_up_function::hdd39238ca1e7535e
39.91 Gc    5.4%	-	 	                   hir_ty::mir::eval::Evaluator::exec_fn_with_args::h5acbc73e38ea6c7e
39.91 Gc    5.4%	-	 	                    hir_ty::mir::eval::shim::_$LT$impl$u20$hir_ty..mir..eval..Evaluator$GT$::detect_and_exec_special_function::h6adaafd39ac2745b
39.91 Gc    5.4%	-	 	                     hir_ty::mir::eval::Evaluator::exec_fn_with_args::h5acbc73e38ea6c7e
39.91 Gc    5.4%	-	 	                      hir_ty::mir::eval::Evaluator::exec_looked_up_function::hdd39238ca1e7535e
39.91 Gc    5.4%	-	 	                       hir_ty::mir::eval::Evaluator::exec_fn_with_args::h5acbc73e38ea6c7e
39.91 Gc    5.4%	-	 	                        hir_ty::mir::eval::shim::_$LT$impl$u20$hir_ty..mir..eval..Evaluator$GT$::detect_and_exec_special_function::h6adaafd39ac2745b
39.91 Gc    5.4%	-	 	                         hir_ty::mir::eval::Evaluator::exec_fn_with_args::h5acbc73e38ea6c7e
39.91 Gc    5.4%	-	 	                          hir_ty::mir::eval::Evaluator::exec_looked_up_function::hdd39238ca1e7535e
39.91 Gc    5.4%	-	 	                           hir_ty::mir::eval::Evaluator::exec_fn_with_args::h5acbc73e38ea6c7e
39.91 Gc    5.4%	-	 	                            hir_ty::mir::eval::shim::_$LT$impl$u20$hir_ty..mir..eval..Evaluator$GT$::detect_and_exec_special_function::h6adaafd39ac2745b
39.91 Gc    5.4%	-	 	                             hir_ty::mir::eval::Evaluator::exec_fn_with_args::h5acbc73e38ea6c7e
39.91 Gc    5.4%	-	 	                              hir_ty::mir::eval::Evaluator::exec_looked_up_function::hdd39238ca1e7535e
39.91 Gc    5.4%	-	 	                               hir_ty::mir::eval::Evaluator::exec_fn_with_args::h5acbc73e38ea6c7e
39.91 Gc    5.4%	-	 	                                hir_ty::mir::eval::shim::_$LT$impl$u20$hir_ty..mir..eval..Evaluator$GT$::detect_and_exec_special_function::h6adaafd39ac2745b
39.91 Gc    5.4%	-	 	                                 hir_ty::mir::eval::Evaluator::exec_fn_with_args::h5acbc73e38ea6c7e
39.91 Gc    5.4%	-	 	                                  hir_ty::mir::eval::Evaluator::exec_looked_up_function::hdd39238ca1e7535e
39.91 Gc    5.4%	-	 	                                   hir_ty::mir::eval::Evaluator::exec_fn_with_args::h5acbc73e38ea6c7e
39.91 Gc    5.4%	-	 	                                    hir_ty::mir::eval::shim::_$LT$impl$u20$hir_ty..mir..eval..Evaluator$GT$::detect_and_exec_special_function::h6adaafd39ac2745b
39.91 Gc    5.4%	-	 	                                     hir_ty::mir::eval::Evaluator::exec_fn_with_args::h5acbc73e38ea6c7e
39.91 Gc    5.4%	-	 	                                      hir_ty::mir::eval::Evaluator::exec_looked_up_function::hdd39238ca1e7535e
39.91 Gc    5.4%	-	 	                                       hir_ty::mir::eval::Evaluator::exec_fn_with_args::h5acbc73e38ea6c7e
39.91 Gc    5.4%	-	 	                                        hir_ty::mir::eval::shim::_$LT$impl$u20$hir_ty..mir..eval..Evaluator$GT$::detect_and_exec_special_function::h6adaafd39ac2745b
39.91 Gc    5.4%	-	 	                                         hir_ty::mir::eval::Evaluator::exec_fn_with_args::h5acbc73e38ea6c7e
39.91 Gc    5.4%	-	 	                                          hir_ty::mir::eval::Evaluator::exec_looked_up_function::hdd39238ca1e7535e
39.91 Gc    5.4%	-	 	                                           hir_ty::mir::eval::Evaluator::exec_fn_with_args::h5acbc73e38ea6c7e
39.91 Gc    5.4%	-	 	                                            hir_ty::mir::eval::shim::_$LT$impl$u20$hir_ty..mir..eval..Evaluator$GT$::detect_and_exec_special_function::h6adaafd39ac2745b
39.91 Gc    5.4%	-	 	                                             hir_ty::mir::eval::Evaluator::exec_fn_with_args::h5acbc73e38ea6c7e
39.91 Gc    5.4%	-	 	                                              hir_ty::mir::eval::Evaluator::exec_looked_up_function::hdd39238ca1e7535e
39.91 Gc    5.4%	-	 	                                               hir_ty::mir::eval::Evaluator::exec_fn_with_args::h5acbc73e38ea6c7e
39.91 Gc    5.4%	-	 	                                                hir_ty::mir::eval::shim::_$LT$impl$u20$hir_ty..mir..eval..Evaluator$GT$::detect_and_exec_special_function::h6adaafd39ac2745b
39.91 Gc    5.4%	-	 	                                                 hir_ty::mir::eval::Evaluator::exec_fn_with_args::h5acbc73e38ea6c7e
39.91 Gc    5.4%	-	 	                                                  hir_ty::mir::eval::Evaluator::exec_looked_up_function::hdd39238ca1e7535e
39.91 Gc    5.4%	-	 	                                                   hir_ty::mir::eval::Evaluator::exec_fn_with_args::h5acbc73e38ea6c7e
39.91 Gc    5.4%	-	 	                                                    hir_ty::mir::eval::shim::_$LT$impl$u20$hir_ty..mir..eval..Evaluator$GT$::detect_and_exec_special_function::h6adaafd39ac2745b
39.91 Gc    5.4%	-	 	                                                     hir_ty::mir::eval::Evaluator::exec_fn_with_args::h5acbc73e38ea6c7e
39.91 Gc    5.4%	-	 	                                                      hir_ty::mir::eval::Evaluator::exec_looked_up_function::hdd39238ca1e7535e
39.91 Gc    5.4%	-	 	                                                       hir_ty::mir::eval::Evaluator::exec_fn_with_args::h5acbc73e38ea6c7e
39.91 Gc    5.4%	-	 	                                                        hir_ty::mir::eval::shim::_$LT$impl$u20$hir_ty..mir..eval..Evaluator$GT$::detect_and_exec_special_function::h6adaafd39ac2745b
39.91 Gc    5.4%	-	 	                                                         hir_ty::mir::eval::Evaluator::exec_fn_with_args::h5acbc73e38ea6c7e
39.91 Gc    5.4%	-	 	                                                          hir_ty::mir::eval::Evaluator::exec_looked_up_function::hdd39238ca1e7535e
39.91 Gc    5.4%	-	 	                                                           hir_ty::mir::eval::Evaluator::exec_fn_with_args::h5acbc73e38ea6c7e
39.91 Gc    5.4%	-	 	                                                            hir_ty::mir::eval::shim::_$LT$impl$u20$hir_ty..mir..eval..Evaluator$GT$::detect_and_exec_special_function::h6adaafd39ac2745b
39.91 Gc    5.4%	-	 	                                                             hir_ty::mir::eval::Evaluator::exec_fn_with_args::h5acbc73e38ea6c7e
39.91 Gc    5.4%	-	 	                                                              hir_ty::mir::eval::Evaluator::exec_looked_up_function::hdd39238ca1e7535e
39.91 Gc    5.4%	-	 	                                                               hir_ty::mir::eval::Evaluator::exec_fn_with_args::h5acbc73e38ea6c7e
39.91 Gc    5.4%	-	 	                                                                hir_ty::mir::eval::shim::_$LT$impl$u20$hir_ty..mir..eval..Evaluator$GT$::detect_and_exec_special_function::h6adaafd39ac2745b
39.91 Gc    5.4%	-	 	                                                                 hir_ty::mir::eval::Evaluator::exec_fn_with_args::h5acbc73e38ea6c7e
39.91 Gc    5.4%	-	 	                                                                  hir_ty::mir::eval::Evaluator::exec_looked_up_function::hdd39238ca1e7535e
39.91 Gc    5.4%	-	 	                                                                   hir_ty::mir::eval::Evaluator::exec_fn_with_args::h5acbc73e38ea6c7e
39.91 Gc    5.4%	-	 	                                                                    hir_ty::mir::eval::shim::_$LT$impl$u20$hir_ty..mir..eval..Evaluator$GT$::detect_and_exec_special_function::h6adaafd39ac2745b
39.91 Gc    5.4%	-	 	                                                                     hir_ty::mir::eval::Evaluator::exec_fn_with_args::h5acbc73e38ea6c7e
39.91 Gc    5.4%	-	 	                                                                      hir_ty::mir::eval::Evaluator::exec_looked_up_function::hdd39238ca1e7535e
39.91 Gc    5.4%	-	 	                                                                       hir_ty::mir::eval::Evaluator::exec_fn_with_args::h5acbc73e38ea6c7e
39.91 Gc    5.4%	-	 	                                                                        hir_ty::mir::eval::shim::_$LT$impl$u20$hir_ty..mir..eval..Evaluator$GT$::detect_and_exec_special_function::h6adaafd39ac2745b
39.91 Gc    5.4%	-	 	                                                                         hir_ty::mir::eval::Evaluator::exec_fn_with_args::h5acbc73e38ea6c7e
39.91 Gc    5.4%	-	 	                                                                          hir_ty::mir::eval::Evaluator::exec_looked_up_function::hdd39238ca1e7535e
39.91 Gc    5.4%	-	 	                                                                           hir_ty::mir::eval::Evaluator::exec_fn_with_args::h5acbc73e38ea6c7e
27.07 Gc    3.7%	-	 	                                                                            hir_ty::mir::eval::interpret_mir::h1aedaa82aa448652
27.07 Gc    3.7%	-	 	                                                                             hir_ty::consteval::const_eval_query::h2ba788d1eeb78fc7
27.07 Gc    3.7%	-	 	                                                                              std::panicking::try::ha56eabb7a167b93e
27.07 Gc    3.7%	-	 	                                                                               salsa::derived::slot::Slot$LT$Q$C$MP$GT$::execute::h5c59ed75e7a434f0
27.07 Gc    3.7%	-	 	                                                                                salsa::derived::slot::Slot$LT$Q$C$MP$GT$::read::he9905a952dc215b5
27.07 Gc    3.7%	-	 	                                                                                 _$LT$salsa..derived..DerivedStorage$LT$Q$C$MP$GT$$u20$as$u20$salsa..plumbing..QueryStorageOps$LT$Q$GT$$GT$::fetch::h8b95ad43432ee1b6
27.07 Gc    3.7%	-	 	                                                                                  _$LT$DB$u20$as$u20$hir_ty..db..HirDatabase$GT$::const_eval::__shim::h584162d92c955f3f
27.07 Gc    3.7%	-	 	                                                                                   hir_ty::mir::lower::MirLowerCtx::lower_const_to_operand::h85e68878893a0834
27.07 Gc    3.7%	-	 	                                                                                    hir_ty::mir::lower::MirLowerCtx::lower_const::h030eeca6c244ea28
27.07 Gc    3.7%	-	 	                                                                                     hir_ty::mir::lower::MirLowerCtx::lower_expr_to_place_without_adjust::hf65a0940cd09e584
27.07 Gc    3.7%	-	 	                                                                                      hir_ty::mir::lower::as_place::_$LT$impl$u20$hir_ty..mir..lower..MirLowerCtx$GT$::lower_expr_to_some_place_without_adjust::h8a830057ba5ed413
27.07 Gc    3.7%	-	 	                                                                                       hir_ty::mir::lower::as_place::_$LT$impl$u20$hir_ty..mir..lower..MirLowerCtx$GT$::lower_expr_as_place_without_adjust::h54ea5d6a709e1caa
27.07 Gc    3.7%	-	 	                                                                                        hir_ty::mir::lower::MirLowerCtx::lower_expr_to_some_operand::h01b4ec732e8a803b
27.07 Gc    3.7%	-	 	                                                                                         hir_ty::mir::lower::MirLowerCtx::lower_expr_to_place_without_adjust::hf65a0940cd09e584
27.07 Gc    3.7%	-	 	                                                                                          hir_ty::mir::lower::MirLowerCtx::lower_block_to_place::h7354a727b7bc6234
27.07 Gc    3.7%	-	 	                                                                                           hir_ty::mir::lower::MirLowerCtx::lower_expr_to_place_without_adjust::hf65a0940cd09e584
27.07 Gc    3.7%	-	 	                                                                                            hir_ty::mir::lower::lower_to_mir::h95472c5151e4b4d2
27.07 Gc    3.7%	-	 	                                                                                             hir_ty::mir::lower::mir_body_query::h18a86d7ad88c976c
27.07 Gc    3.7%	-	 	                                                                                              salsa::Cycle::catch::hef43b308ad311631
27.07 Gc    3.7%	-	 	                                                                                               salsa::derived::slot::Slot$LT$Q$C$MP$GT$::execute::h1e292714cd890b1e
27.07 Gc    3.7%	-	 	                                                                                                salsa::derived::slot::Slot$LT$Q$C$MP$GT$::read::hc568641904c9d3ea
27.07 Gc    3.7%	-	 	                                                                                                 _$LT$salsa..derived..DerivedStorage$LT$Q$C$MP$GT$$u20$as$u20$salsa..plumbing..QueryStorageOps$LT$Q$GT$$GT$::fetch::hf1744aa5ed8e8bb6
27.07 Gc    3.7%	-	 	                                                                                                  _$LT$DB$u20$as$u20$hir_ty..db..HirDatabase$GT$::mir_body::ha0af715c0b7dd401
27.07 Gc    3.7%	-	 	                                                                                                   hir_ty::mir::monomorphization::monomorphized_mir_body_query::hb61fe31c91a2cdf8
27.07 Gc    3.7%	-	 	                                                                                                    std::panicking::try::hd5303536050a8743
27.07 Gc    3.7%	-	 	                                                                                                     salsa::derived::slot::Slot$LT$Q$C$MP$GT$::execute::h334103e5b06d0694
27.07 Gc    3.7%	-	 	                                                                                                      salsa::derived::slot::Slot$LT$Q$C$MP$GT$::read::h15457b97840b3589
27.07 Gc    3.7%	-	 	                                                                                                       _$LT$salsa..derived..DerivedStorage$LT$Q$C$MP$GT$$u20$as$u20$salsa..plumbing..QueryStorageOps$LT$Q$GT$$GT$::fetch::h50e90ed47fdf94f3
27.07 Gc    3.7%	-	 	                                                                                                        _$LT$DB$u20$as$u20$hir_ty..db..HirDatabase$GT$::monomorphized_mir_body::__shim::h5566874ab84f7d74
27.07 Gc    3.7%	-	 	                                                                                                         hir_ty::mir::eval::Evaluator::get_mir_or_dyn_index::h45b67618cec354d2
27.07 Gc    3.7%	-	 	                                                                                                          hir_ty::mir::eval::Evaluator::exec_fn_with_args::h5acbc73e38ea6c7e
27.07 Gc    3.7%	-	 	                                                                                                           hir_ty::mir::eval::interpret_mir::h1aedaa82aa448652
27.07 Gc    3.7%	-	 	                                                                                                            hir_ty::consteval::const_eval_query::h2ba788d1eeb78fc7
27.07 Gc    3.7%	-	 	                                                                                                             std::panicking::try::ha56eabb7a167b93e
27.07 Gc    3.7%	-	 	                                                                                                              salsa::derived::slot::Slot$LT$Q$C$MP$GT$::execute::h5c59ed75e7a434f0
27.07 Gc    3.7%	-	 	                                                                                                               salsa::derived::slot::Slot$LT$Q$C$MP$GT$::read::he9905a952dc215b5
27.07 Gc    3.7%	-	 	                                                                                                                _$LT$salsa..derived..DerivedStorage$LT$Q$C$MP$GT$$u20$as$u20$salsa..plumbing..QueryStorageOps$LT$Q$GT$$GT$::fetch::h8b95ad43432ee1b6
27.07 Gc    3.7%	-	 	                                                                                                                 _$LT$DB$u20$as$u20$hir_ty..db..HirDatabase$GT$::const_eval::__shim::h584162d92c955f3f
27.07 Gc    3.7%	-	 	                                                                                                                  hir_ty::mir::lower::MirLowerCtx::lower_const_to_operand::h85e68878893a0834
27.07 Gc    3.7%	-	 	                                                                                                                   hir_ty::mir::lower::MirLowerCtx::lower_const::h030eeca6c244ea28
27.07 Gc    3.7%	-	 	                                                                                                                    hir_ty::mir::lower::MirLowerCtx::lower_expr_to_place_without_adjust::hf65a0940cd09e584
27.07 Gc    3.7%	-	 	                                                                                                                     hir_ty::mir::lower::as_place::_$LT$impl$u20$hir_ty..mir..lower..MirLowerCtx$GT$::lower_expr_to_some_place_without_adjust::h8a830057ba5ed413
27.07 Gc    3.7%	-	 	                                                                                                                      hir_ty::mir::lower::as_place::_$LT$impl$u20$hir_ty..mir..lower..MirLowerCtx$GT$::lower_expr_as_place_without_adjust::h54ea5d6a709e1caa
27.07 Gc    3.7%	-	 	                                                                                                                       hir_ty::mir::lower::MirLowerCtx::lower_expr_to_some_operand::h01b4ec732e8a803b
27.07 Gc    3.7%	-	 	                                                                                                                        hir_ty::mir::lower::MirLowerCtx::lower_expr_to_place_without_adjust::hf65a0940cd09e584
27.07 Gc    3.7%	-	 	                                                                                                                         hir_ty::mir::lower::as_place::_$LT$impl$u20$hir_ty..mir..lower..MirLowerCtx$GT$::lower_expr_to_some_place_without_adjust::h8a830057ba5ed413
27.07 Gc    3.7%	-	 	                                                                                                                          hir_ty::mir::lower::MirLowerCtx::lower_block_to_place::h7354a727b7bc6234
27.07 Gc    3.7%	-	 	                                                                                                                           hir_ty::mir::lower::MirLowerCtx::lower_expr_to_place_without_adjust::hf65a0940cd09e584
27.07 Gc    3.7%	-	 	                                                                                                                            hir_ty::mir::lower::as_place::_$LT$impl$u20$hir_ty..mir..lower..MirLowerCtx$GT$::lower_expr_to_some_place_without_adjust::h8a830057ba5ed413
27.07 Gc    3.7%	-	 	                                                                                                                             hir_ty::mir::lower::MirLowerCtx::lower_block_to_place::h7354a727b7bc6234
27.07 Gc    3.7%	-	 	                                                                                                                              hir_ty::mir::lower::MirLowerCtx::lower_expr_to_place_without_adjust::hf65a0940cd09e584
27.07 Gc    3.7%	-	 	                                                                                                                               hir_ty::mir::lower::lower_to_mir::h95472c5151e4b4d2
27.07 Gc    3.7%	-	 	                                                                                                                                hir_ty::mir::lower::mir_body_query::h18a86d7ad88c976c
27.07 Gc    3.7%	-	 	                                                                                                                                 salsa::Cycle::catch::hef43b308ad311631
27.07 Gc    3.7%	-	 	                                                                                                                                  salsa::derived::slot::Slot$LT$Q$C$MP$GT$::execute::h1e292714cd890b1e
27.07 Gc    3.7%	-	 	                                                                                                                                   salsa::derived::slot::Slot$LT$Q$C$MP$GT$::read::hc568641904c9d3ea
27.07 Gc    3.7%	-	 	                                                                                                                                    _$LT$salsa..derived..DerivedStorage$LT$Q$C$MP$GT$$u20$as$u20$salsa..plumbing..QueryStorageOps$LT$Q$GT$$GT$::fetch::hf1744aa5ed8e8bb6
27.07 Gc    3.7%	-	 	                                                                                                                                     _$LT$DB$u20$as$u20$hir_ty..db..HirDatabase$GT$::mir_body::ha0af715c0b7dd401
27.07 Gc    3.7%	-	 	                                                                                                                                      hir_ty::mir::borrowck::borrowck_query::h6e8ab84e090da7d1
27.07 Gc    3.7%	-	 	                                                                                                                                       salsa::Cycle::catch::h7c7efa842d4f3b4a
27.07 Gc    3.7%	-	 	                                                                                                                                        salsa::derived::slot::Slot$LT$Q$C$MP$GT$::execute::hd56387d6dbab459a
27.07 Gc    3.7%	-	 	                                                                                                                                         salsa::derived::slot::Slot$LT$Q$C$MP$GT$::read::hd6b9b2dbbf79533d
27.07 Gc    3.7%	-	 	                                                                                                                                          _$LT$salsa..derived..DerivedStorage$LT$Q$C$MP$GT$$u20$as$u20$salsa..plumbing..QueryStorageOps$LT$Q$GT$$GT$::fetch::h7ec1a47af173b4c2
27.07 Gc    3.7%	-	 	                                                                                                                                           _$LT$DB$u20$as$u20$hir_ty..db..HirDatabase$GT$::borrowck::h79fb90646fa78b1c
27.07 Gc    3.7%	-	 	                                                                                                                                            hir::DefWithBody::diagnostics::hc65b742f71c5b2cc
27.07 Gc    3.7%	-	 	                                                                                                                                             hir::Module::diagnostics::h62ac7645837d2d56
27.07 Gc    3.7%	-	 	                                                                                                                                              ide_diagnostics::diagnostics::hede181c86305aab1
27.07 Gc    3.7%	-	 	                                                                                                                                               salsa::Cancelled::catch::hcb21db1b01fd9e2a
27.07 Gc    3.7%	-	 	                                                                                                                                                core::ops::function::impls::_$LT$impl$u20$core..ops..function..FnMut$LT$A$GT$$u20$for$u20$$RF$mut$u20$F$GT$::call_mut::hdcf8d4c90360f1ff
```

It hapepens in both Sublime and VSCode. VSCode only logs:

```
ERROR hir::diagnostics: synthetic syntax
```

**Comment by Nilstrieb** [comment]
yeah, should be fixed by #16935

**Comment by kornelski** [comment]
Yes, that's probably it, because downgrade to last-good version mentioned in these issues works for me.
